controllers/storageautoscaler/scraper.go

```diff
@@ -107,9 +112,11 @@ func filterObjectsForScaling(storageAutoScalerList *ocsv1.StorageAutoScalerList,
 			err := fmt.Errorf("osd usage is nil for device class, device class: %s", storageAutoScaler.Spec.DeviceClass)
 			log.Error(err, "device class not found in sync map")
 		} else {
-			deviceClassUsage := usage.(float64)
-			if (deviceClassUsage * 100) >= float64(deviceClassThreshold) {
-				objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+			deviceClassUsage := usage.(OsdUsage)
+			if deviceClassUsage.TotalUsage*100 >= math.Max(float64(deviceClassThreshold)-10, 0) {
+				if (deviceClassUsage.HighestUsage * 100) >= float64(deviceClassThreshold) {
```

Comment:
nit: It may be helpful to log if the highest usage OSD is above the threshold, but we are skipping the expansion. So you may want to switch the order of these checks, and log in that case?